subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/NonHierarchicalFileWatcherUpdater.java

```diff
@@ -146,7 +156,9 @@ private void updateWatchedDirectories(Map<String, Integer> changedWatchDirectori
         LOGGER.info("Watching {} directories to track changes", watchedDirectories.entrySet().size());
         try {
             if (!directoriesToStopWatching.isEmpty()) {
-                fileWatcher.stopWatching(directoriesToStopWatching);
+                if (!fileWatcher.stopWatching(directoriesToStopWatching)) {
+                    LOGGER.warn("Couldn't stop watching directories: {}", directoriesToStopWatching);
```

Comment:
Should we keep this? Should it be at `warn`?

Reply:
I think we receive `false` here if a directory already isn't watched for example because it was deleted, right? So I'd say let's remove it or log it at `debug`.